setup.py

```diff
@@ -78,18 +79,44 @@ def read_response(cmd):
 
     except Exception as e:
         if platform.system() == "Windows":
-            # try to get GDAL version from command line
-            # Note: additional command-line parameters required to point to GDAL
-            if not "GDAL_VERSION":
-                raise ValueError(
-                    "GDAL_VERSION must be provided as an environment variable"
-                )
-            GDAL_VERSION = tuple(
-                int(i) for i in os.environ.get("GDAL_VERSION", "").split(".")
-            )
+            # Note: additional command-line parameters required to point to GDAL;
+            # see the README.
+
+            gdal_version_str = os.environ.get("GDAL_VERSION", "")
+            if not gdal_version_str:
+                # attempt to execute gdalinfo to find GDAL version
+                # Note: gdalinfo must be on the PATH
+                try:
+                    gdalinfo_path = None
+                    for path in os.getenv("PATH", "").split(os.pathsep):
+                        matches = list(Path(path).glob("**/gdalinfo*"))
+                        if matches:
+                            gdalinfo_path = matches[0]
+                            break
+
+                    if gdalinfo_path:
+                        print(f"Found gdalinfo at {gdalinfo_path}")
+                        raw_version = read_response([gdalinfo_path, "--version"]) or ""
+                        # gdal_version_str = raw_version.split(',')[0].split(' ')[1]
+                        m = re.search("\d+\.\d+\.\d+", raw_version)
+                        if m:
+                            gdal_version_str = m.group()
+                    else:
+                        print("Unable to find gdalinfo")
```

Comment:
just a question: do you have some rationale for when to use print and when to use logging?

Reply:
Oops - these were for debugging that I forgot to remove before merging.

For logging in setup.py I was roughly following what Fiona was doing.